qf_lib/analysis/signals_analysis/signals_plotter.py

```diff
@@ -233,7 +233,11 @@ def _get_signals_dates(self):
                 start_time = day + MarketOpenEvent.trigger_time()
                 end_time = day + MarketCloseEvent.trigger_time()
                 signal_times_for_a_day = date_range(start_time, end_time, freq=self.signal_frequency.to_pandas_freq())
+
                 signal_time_holder.append(signal_times_for_a_day)
-            evaluation_dates = signal_time_holder[0].union_many(signal_time_holder[1:])
+
+            evaluation_dates = signal_time_holder[0]
+            for idx in signal_time_holder[1:]:
+                evaluation_dates = evaluation_dates.union(idx)
```

Comment:
here I'm pretty sure you can just do `set().union(*signal_time_holder)`

Reply:
The result is different.

This is changed due to the removal of pandas union_many function.

One could do the following:
```
evaluation_dates= set().union(*signal_time_holder)
evaluation_dates= DatetimeIndex(sorted(list(evaluation_dates)))
```

In case of increasing len(evaluation_dates) and very short date ranges the solution with set is performing better. However if the date_ranges spans over more than a year - the original solution is performing significantly better.